src/Api/Data/DbTask.cs

```diff
@@ -6,19 +6,26 @@
 using Microsoft.Extensions.DependencyInjection;
 using Api.Models.Entity;
 using System;
+using Microsoft.AspNetCore.Hosting;
 
 namespace Api.Data
 {
     public static class DbTask
     {
-        public static void RunMigrationsAndSeedDb(this IApplicationBuilder app)
+        public static void RunMigrationsAndSeedDb(this IApplicationBuilder app, IHostingEnvironment env)
         {
             var dbContext = app.ApplicationServices.GetRequiredService<CmsDbContext>();
             var appConfig = app.ApplicationServices.GetRequiredService<AppConfig>();
             var userManager = new UserManager(dbContext);
 
-            // Run Migrations to apply any pending migrations. (if any)
-            dbContext.Database.Migrate();
+            //Basically we need this because the testing library is using an in-memory database and hence it does not needs that call. 
+            //Therefore we need a check checking if the environment is development or Test.
+            //The default environment on which the tests are executed is 'Test'.
+            if (env.IsDevelopment())
+            {
+                // Run Migrations to apply any pending migrations. (if any)
+                dbContext.Database.Migrate();
```

Comment:
No. IsDevelopment is not related to the Test environment. We need the Migrations primary on the productive enviorment.